pkg/agent/cluster.go

```diff
@@ -0,0 +1,332 @@
+		}
+
+		configmap, err := czero.API.Kube.IsBootstrapConfigMapComplete()
+		if configmap {
+			logrus.Info("bootstrap configMap status is complete")
+			return true, nil
+		}
+		if err != nil {
+			logrus.Debug(err)
+		}
+	}
+
+	agentRestAPILive, agentRestAPIErr := czero.API.Rest.IsRestAPILive()
+	if agentRestAPIErr != nil {
+		logrus.Trace(errors.Wrap(agentRestAPIErr, "Agent Rest API is not available"))
+	}
+
+	if agentRestAPILive {
+
+		// First time we see the agent Rest API
+		if !czero.installHistory.RestAPISeen {
+			logrus.Debug("Agent Rest API Initialized")
+			czero.installHistory.RestAPISeen = true
+		}
+
+		// Lazy loading of the clusterID and clusterInfraEnvID
+		if czero.clusterID == nil {
+			clusterID, err := czero.API.Rest.getClusterID()
+			if err != nil {
+				return false, errors.Wrap(err, "unable to retrieve clusterID from Agent Rest API")
+			}
+			czero.clusterID = clusterID
+		}
+
+		if czero.clusterInfraEnvID == nil {
+			clusterInfraEnvID, err := czero.API.Rest.getClusterInfraEnvID()
+			if err != nil {
+				return false, errors.Wrap(err, "unable to retrieve clusterInfraEnvID from Agent Rest API")
+			}
+			czero.clusterInfraEnvID = clusterInfraEnvID
+		}
+
+		logrus.Trace("getting cluster metadata from Agent Rest API")
+		clusterMetadata, err := czero.GetClusterRestAPIMetadata()
+		if err != nil {
+			return false, errors.Wrap(err, "unable to retrieve cluster metadata from Agent Rest API")
+		}
+
+		if clusterMetadata == nil {
+			return false, errors.New("cluster metadata returned nil from Agent Rest API")
+		}
+
+		czero.PrintInstallStatus(clusterMetadata)
+		czero.installHistory.RestAPIPreviousClusterStatus = *clusterMetadata.Status
+
+		// Update Install History object when we see these states
+		czero.updateInstallHistoryClusterStatus(clusterMetadata)
+
+		installing, _ := czero.IsInstalling(*clusterMetadata.Status)
+		if !installing {
+			logrus.Warn("Cluster has stopped installing... working to recover installation")
+			errored, _ := czero.HasErrored(*clusterMetadata.Status)
+			if errored {
+				return false, errors.New("cluster installation has stopped due to errors")
+			} else if *clusterMetadata.Status == models.ClusterStatusCancelled {
+				return false, errors.New("cluster installation was cancelled")
+			}
+		}
+
+		// Print most recent event assoicated with the clusterInfraEnvID
+		eventList, err := czero.API.Rest.GetInfraEnvEvents(czero.clusterInfraEnvID)
+		if err != nil {
+			return false, errors.Wrap(err, "unable to retrieve events about the cluster from the Agent Rest API")
+		}
+		if len(eventList) == 0 {
+			logrus.Trace("no cluster events detected from the Agent Rest API")
+		} else {
+			mostRecentEvent := eventList[len(eventList)-1]
+			// Don't print the same status message back to back
+			if mostRecentEvent.Message != &czero.installHistory.RestAPIPreviousEventMessage {
+				if *mostRecentEvent.Severity == models.EventSeverityInfo {
+					logrus.Info(mostRecentEvent.Message)
```

Comment:
```suggestion
					logrus.Info(*mostRecentEvent.Message)
```
(otherwise the pointer address gets printed instead of the message)